nf_core/download.py

```diff
@@ -167,95 +166,187 @@ def download_workflow(self):
+                    break
+
+            # Can't find the release or branch - throw an error
+            else:
+                log.info(
+                    "Available {} releases: '{}'".format(
+                        self.pipeline, "', '".join([r["tag_name"] for r in self.wf_releases])
+                    )
+                )
+                log.info("Available {} branches: '{}'".format(self.pipeline, "', '".join(self.wf_branches.keys())))
+                raise AssertionError(
+                    "Not able to find release / branch '{}' for {}".format(self.release, self.pipeline)
+                )
+
+        # Set the outdir
+        if not self.outdir:
+            self.outdir = "{}-{}".format(self.pipeline.replace("/", "-").lower(), self.release)
+
+        # Set the download URL and return
+        self.wf_download_url = "https://github.com/{}/archive/{}.zip".format(self.pipeline, self.wf_sha)
+
+    def prompt_container_download(self):
+        """Prompt whether to download container images or not"""
+
+        if self.container is None:
+            stderr.print("\nIn addition to the pipeline code, this tool can download software containers.")
+            self.container = questionary.select(
+                "Download software container images:",
+                choices=["none", "singularity"],
+                style=nf_core.utils.nfcore_question_style,
+            ).unsafe_ask()
+
+    def prompt_use_singularity_cachedir(self):
+        """Prompt about using $NXF_SINGULARITY_CACHEDIR if not already set"""
+        if (
+            self.container == "singularity"
+            and os.environ.get("NXF_SINGULARITY_CACHEDIR") is None
+            and stderr.is_interactive  # Use rich auto-detection of interactive shells
+        ):
+            stderr.print(
+                "\nNextflow and nf-core can use an environment variable called [blue]$NXF_SINGULARITY_CACHEDIR[/] that is a path to a directory where remote Singularity images are stored. "
+                "This allows downloaded images to be cached in a central location."
+            )
+            if rich.prompt.Confirm.ask(
+                f"[blue bold]?[/] [bold]Define [blue not bold]$NXF_SINGULARITY_CACHEDIR[/] for a shared Singularity image download folder?[/]"
+            ):
+                # Prompt user for a cache directory path
+                cachedir_path = None
+                while cachedir_path is None:
+                    cachedir_path = os.path.abspath(
+                        questionary.path(
+                            "Specify the path:", only_directories=True, style=nf_core.utils.nfcore_question_style
+                        ).unsafe_ask()
+                    )
+                    if cachedir_path == os.path.abspath(""):
+                        log.error(f"Not using [blue]$NXF_SINGULARITY_CACHEDIR[/]")
+                        cachedir_path = False
+                    elif not os.path.isdir(cachedir_path):
```

Comment:
Somehow I always end up there, even if clearly specifying a directory ...

Reply:
It works as I believe intended if specifying an existing directory.

But a better behaviour might be to offer to try to create it if it doesn't exist.

Reply:
for me it also broke when using an existing directory :shrug: will try again later .... I had some issue with `os.path.isdir` before

Reply:
Marked as resolved - can I safely ignore this? I've not been able to reproduce any problems with it..